www/classes/Util/Singleton/Configuration.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Util\Singleton;
+
+use Util\Singleton\Database;
+
+/**
+ * Model voor de single variables.
+ */
+class Configuration
+{
+    /** @var Configuration|null Singleton instance. */
+    private static ?Configuration $instance = null;
+
+    /** @var string|null */
+    protected string|null $schrijfregels;
+
+    /** @var int|null */
+    protected int|null $min_checks;
+
+    /** @var string|null */
+    protected string|null $mail_address;
+
+    /** @var (string|null)[]|null */
+    protected array|null $passwords;
+
+    /**
+     * Returns the singleton instance.
+     * @return Configuration
+     */
+    public static function instance(): Configuration
+    {
+        if (self::$instance === null) {
+            self::$instance = new Configuration();
+        }
+        return self::$instance;
+    }
+
+    public function __construct()
+    {
+        $this->schrijfregels = null;
+        $this->min_checks = null;
+        $this->mail_address = null;
+        $this->passwords = null;
+    }
+
+    /**
+     * @param $name
+     * @return string|null|array|int
+     * */
+    public function __get($name)
+    {
+        if ($this->$name == null) {
+            Database::instance()->storeQuery("SELECT * FROM configuration WHERE name = ?");
+            $stmt = Database::instance()->prepareStoredQuery();
+            $stmt->bind_param(
+                's',
+                $name
+            );
+            $stmt->execute();
+            $config_data = $stmt->get_result()->fetch_assoc();
+            $result = $config_data ? $config_data['value'] : null;
+        } else {
+            return $this->$name;
+        }
+        switch ($name) {
+            case "min_checks":
+                $this->min_checks = (int)$result;
+                break;
+            case "passwords":
+                $this->passwords = explode(",", $result);
+                $this->passwords[1] = $this->passwords[1] == "" ? null : $this->passwords[1];
+                $this->passwords[2] = $this->passwords[2] == "" ? null : $this->passwords[2];
+                $this->passwords[3] = $this->passwords[3] == "" ? null : $this->passwords[3];
+                break;
+            default:
+                $this->$name = $result;
```

Comment:
Dit is ook wel risky qua fouten, liever gewoon alleen de vastgestelde velden toelaten.